### PR TITLE
Add UserValidation during login

### PR DESCRIPTION
**Pull request by sBouzols**
Signed-off-by: sBouzols <sylvain.bouzols@gmail.com>

### CODE REVIEW

File: src/components/AuthenticationRouter/AuthenticationRouter.js [hunk 1]
```diff
@@ -33,50 +38,69 @@ const AuthenticationRouter = ({
...
-                                disabled={userManager.instance === null}
-                                onLoginClick={() =>
-                                    login(location, userManager.instance)
+            <Grid container alignContent={'center'} direction={'column'}>
+                {userManager.error !== null && (
+                    <h1>Error : Getting userManager; {userManager.error}</h1>
+                )}
+                {signInCallbackError !== null && (
+                    <h1>
+                        Error : SignIn Callback Error;{' '}
```
Comment (jonenst) [inline src/components/AuthenticationRouter/AuthenticationRouter.js:47]: spurious {' '}

File: src/components/AuthenticationRouter/AuthenticationRouter.js [hunk 2]
```diff
@@ -33,50 +38,69 @@ const AuthenticationRouter = ({
...
+                                    onLoginClick={() =>
+                                        login(location, userManager.instance)
+                                    }
+                                />
+                            )
+                        }
+                    />
+                </Routes>
+
+                {userManager.warning !== null && (
```
Comment (jonenst) [inline src/components/AuthenticationRouter/AuthenticationRouter.js:93]: to group with other error cases above ?

File: src/utils/AuthService.js [hunk 3]
```diff
@@ -205,10 +233,12 @@ function handleSilentRenewCallback(userManagerInstance) {
...
 }
 
-function handleUser(dispatch, userManager) {
+function handleUser(dispatch, validateUser, userManager) {
     userManager.events.addUserLoaded((user) => {
-        console.debug('user loaded');
-        dispatchUser(dispatch, userManager);
+        console.debug('user loaded', user);
+        dispatchUser(dispatch, validateUser, userManager).catch((e) => {
+            console.log('error in user loaded validation :  ignored');
```
Comment (jonenst) [inline src/utils/AuthService.js:240]: to remove ?

File: src/components/AuthenticationRouter/AuthenticationRouter.js (hunk 1 above)
Comment (sBouzols) [reply src/components/AuthenticationRouter/AuthenticationRouter.js]: removed

File: src/components/AuthenticationRouter/AuthenticationRouter.js (hunk 2 above)
Comment (sBouzols) [reply src/components/AuthenticationRouter/AuthenticationRouter.js]: outdated

File: src/utils/AuthService.js (hunk 3 above)
Comment (sBouzols) [reply src/utils/AuthService.js]: removed

**Review by archendel — APPROVED** [review]
Test ok, excepct it's not easy for an unauthorized user to logout after receiving the error message

**Review by jonenst — APPROVED** [review]

**Pull request merged**